scripts/build_test.py

```diff
@@ -647,4 +648,35 @@ def test_generate_delete_tasks_to_remove_deleted_files(self):
         self.assertEqual(
             len(delete_tasks), build.get_file_count(MOCK_TEMPLATES_DEV_DIR))
 
+    def test_compiled_js_dir_is_deleted_before_compilation(self):
+        """Test that compiled_js_dir is deleted before a fresh
+        compilation.
+        """
+        # pylint: disable=unused-argument
+        def mock_check_call(cmd):
+            pass
+        def mock_call(cmd, shell, stdout):
+            pass
+        def mock_popen(cmd, stdout):
+            pass
+        # pylint: enable=unused-argument
+
+        compiled_js_dir = os.path.join(
+            'core', 'tests', 'build_sources', 'compiled_js_dir', '')
```

Comment:
Ideally make use of the constants in build.py (see above comments).

Reply:
Done.